### PR TITLE
Add support for restricting display area

### PR DESCRIPTION
**Pull request by mmabey**
Sometimes it can be really useful to change the areas of the screen that the buffer is written to, so I added these functions. Calling them without parameters will reset to the default of the whole screen.

### CODE REVIEW

File: ssd1306/ssd1306.py [hunk 1]
```diff
@@ -185,3 +185,26 @@ def contrast(self, contrast):
...
     self.write_command(contrast)
 
+  def set_start_end_cols(self, start_col=0, end_col=None):
+    if end_col is None:
+      end_col = self.columns - 1
+    if start_col < 0 or start_col > self.columns - 1:
+      raise ValueError('Start column must be between 0 and %d.' % (self.columns - 1,))
+    if end_col < 0 or end_col > self.columns -1:
+      raise ValueError('End column must be between 0 and %d.' % (self.columns - 1,))
+
```
Comment (khenderick) [inline ssd1306/ssd1306.py]: Is it possible to add another check to make sure that ```start_col``` is smaller than ```end_col```, or is this something that is possible? If that would be the case, what's the behavior?

The same for ```start_page``` and ```end_page``` below.

Comment (mmabey) [inline ssd1306/ssd1306.py]: The start/end location for pages and columns is an inclusive range. For example, if you call `set_start_end_cols(0, 0)` you'll write to only column zero. How this actually writes to the screen depends on the addressing mode (see pgs. 35-36 of the [datasheet](https://www.adafruit.com/datasheets/SSD1306.pdf)), but the bottom line is that it doesn't have to be a strict inequality.

But you're definitely right, and I appreciate you pointing that out. I'll add another check to make sure that `end_page` isn't less than `start_page` and same for the columns.

**Comment by mmabey** [comment]
@khenderick do you see any other issues with PR #4?

**Comment by khenderick** [comment]
I'm so sorry, seems I missed that last commit. Seems fine to me.

**Pull request merged**